File: mini_proyecto_cuenta_bancaria/bank_account.py
from transaction import Transaction

class BankAccount:
    total_cuentas = 0

    def __init__(self, numero_cuenta, saldo_inicial):
        self.numero_cuenta = numero_cuenta
        self.saldo = saldo_inicial
        self.transactions = []
        BankAccount.total_cuentas += 1

    def add_transaction(self, t):
        self.transactions.append(t)
        self.saldo += t.monto

    def display(self):
        print(f"Cuenta: {self.numero_cuenta}, Saldo: {self.saldo}")
        print("Transacciones:")
        for t in self.transactions:
            t.display()

File: mini_proyecto_cuenta_bancaria/BankAccount.java
import java.util.ArrayList;
import java.util.List;

public class BankAccount {
    private String numeroCuenta;
    private double saldo;
    private List<Transaction> transactions;
    public static int totalCuentas = 0;

    public BankAccount(String numeroCuenta, double saldoInicial) {
        this.numeroCuenta = numeroCuenta;
        this.saldo = saldoInicial;
        this.transactions = new ArrayList<>();
        totalCuentas++;
    }

    public void addTransaction(Transaction t) {
        transactions.add(t);
        saldo += t.getMonto();
    }

    public void display() {
        System.out.println("Cuenta: " + numeroCuenta + ", Saldo: " + saldo);
        System.out.println("Transacciones:");
        for (Transaction t : transactions) {
            t.display();
        }
    }
}